src/components/wallet/methods/depositAsset.ts

```diff
@@ -157,40 +153,89 @@ export default async function depositAsset(
-        this.updateAccount()
-        this.logger.instruction('Transaction status complete')
-        finish()
-      } else {
-        this.logger.instruction('Transaction status pending...')
-        setTimeout(() => {
-          const urlBuilder = new URL(transaction.more_info_url)
-          urlBuilder.searchParams.set('callback', 'postMessage')
-
-          popup.location.href = urlBuilder.toString()
-        }, 1000)
+    let currentStatus = 'incomplete'
+    const transactionUrl = new URL(
+      `${toml.TRANSFER_SERVER_SEP0024}/transaction?id=${interactiveJson.id}`
+    )
+    this.logger.instruction(`Polling for updates: ${transactionUrl.toString()}`)
+    while (!popup.closed && !['completed', 'error'].includes(currentStatus)) {
+      response = await fetch(transactionUrl.toString(), {
+        headers: { Authorization: `Bearer ${auth}` },
+      })
+      let transactionJson = await response.json()
+      if (transactionJson.transaction.status !== currentStatus) {
+        currentStatus = transactionJson.transaction.status
+        popup.location.href = transactionJson.transaction.more_info_url
+        this.logger.instruction(
+          `Transaction ${interactiveJson.id} is in ${transactionJson.transaction.status} status`
+        )
+        switch (currentStatus) {
+          case 'pending_user_transfer_start': {
+            this.logger.instruction(
+              'The anchor is waiting on you to take the action described in the popup'
+            )
+            break
+          }
+          case 'pending_anchor': {
+            this.logger.instruction('The anchor is processing the transaction')
+            break
+          }
+          case 'pending_stellar': {
+            this.logger.instruction(
+              'The Stellar network is processing the transaction'
+            )
+            break
+          }
+          case 'pending_external': {
+            this.logger.instruction(
+              'The transaction is being processed by an external system'
+            )
+            break
+          }
+          case 'pending_trust': {
+            this.logger.instruction(
+              'You must add a trustline to the asset in order to receive your deposit'
+            )
+            this.logger.instruction('Adding trustline...')
+            await this.trustAsset(asset_code, asset_issuer)
+            break
+          }
+          case 'pending_user': {
+            this.logger.instruction(
+              'The anchor is waiting for you to take the action described in the popup'
+            )
+            break
+          }
+          case 'error': {
+            this.logger.instruction(
+              'There was a problem processing your transaction'
+            )
+            break
+          }
+        }
       }
+      // run loop every 2 seconds
+      await new Promise((resolve) => setTimeout(resolve, 2000))
     }
+
+    this.updateAccount()
+    this.logger.instruction('Transaction status complete')
```

Comment:
Before communicating that the transaction is complete I believe we should check if the `currentStatus` is `complete` and if `popup.closed`.
If I were to close the pop up before its complete the logger states that it is.

Reply:
Done!